torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1653,6 +1675,7 @@ std::vector<std::shared_ptr<NCCLComm>>& ProcessGroupNCCL::getNCCLComm(
   std::lock_guard<std::mutex> lock(mutex_);
 
   // Record the communicators based on ncclUniqueId.
+  deviceComms_ = ncclComms;
```

Comment:
`deviceComms_` seems to point to the most-recently created comms. Is that the intention? Or is the intention to point to the global comm (the first-created one)?

Reply:
ooh, good catch